components/new-contribution-flow/StepPayment.js

```diff
@@ -1,16 +1,283 @@
-  static propTypes = {
-    collective: PropTypes.object,
-  };
+import Container from '../../components/Container';
+import { Box, Flex } from '../../components/Grid';
+import CreditCardInactive from '../../components/icons/CreditCardInactive';
+import Loading from '../../components/Loading';
+import MessageBox from '../../components/MessageBox';
+import NewCreditCardForm from '../../components/NewCreditCardForm';
+import { withStripeLoader } from '../../components/StripeProvider';
+import StyledButton from '../../components/StyledButton';
+import StyledHr from '../../components/StyledHr';
+import StyledRadioList from '../../components/StyledRadioList';
+import { P } from '../../components/Text';
+import { withUser } from '../../components/UserProvider';
+
+const PaymentMethodBox = styled(Container)`
+  border-top: 1px solid ${themeGet('colors.black.300')};
+  display: flex;
+  flex-direction: column;
+  background: ${themeGet('colors.white.full')};
+  &:hover {
+    background: ${themeGet('colors.black.50')};
+  }
+`;
 
-  render() {
-    return <Box width={1}>Payment method chooser - same for all states</Box>;
+const paymentMethodsQuery = gqlV2/* GraphQL */ `
+  query UpdatePaymentMethodPopUpPaymentMethod($slug: String) {
+    account(slug: $slug) {
+      id
+      paymentMethods(types: ["creditcard", "virtualcard", "prepaid"]) {
+        id
+        name
+        data
+        service
+        type
+        balance {
+          value
+          currency
+        }
+        account {
+          id
+        }
+      }
+    }
   }
-}
+`;
+
+const NewContributionFlowStepPayment = ({
+  LoggedInUser,
+  stepDetails,
+  stepProfile,
+  stepPayment,
+  collective,
+  onChange,
+  loadStripe,
+}) => {
+  const [loadingSelectedPaymentMethod, setLoadingSelectedPaymentMethod] = useState(true);
+  const [noPaymentMethods, setNoPaymentMethods] = useState(false);
+  const [stripeIsReady, setStripeIsReady] = useState(false);
+  const [stripe, setStripe] = useState(null);
+  const [paymentObject, setPaymentObject] = useState(null);
+
+  const connectedAccounts = get(collective, 'host.connectedAccounts', null);
+  const hostHasStripe = Boolean(find(connectedAccounts, ['service', 'stripe']));
+  const withPaypal =
+    !get(stepDetails, 'interval') &&
+    connectedAccounts &&
+    Boolean(connectedAccounts.find(ca => ca.service === 'stripe'));
+  const withManual = get(collective, 'host.settings.paymentMethods.manual', null);
+
+  console.log('step payment', stepPayment);
+  console.log('sstepProfile', stepProfile);
+
+  const slugForQuery = stepProfile.isIncognito ? LoggedInUser.collective.slug : stepProfile.slug;
+
+  // GraphQL mutations and queries
+  const { data } = useQuery(paymentMethodsQuery, {
+    variables: {
+      slug: slugForQuery,
+    },
+    context: API_V2_CONTEXT,
+  });
+
+  // load stripe on mount
+  useEffect(() => {
+    loadStripe();
+    setStripeIsReady(true);
+  }, [stripeIsReady]);
```

Comment:
If the goal is to trigger it only one time when mounting, you can simply do

```suggestion
  useEffect(() => {
    loadStripe();
  }, []);
```

Also it would be nice to only load stripe if it's supported, so:
```suggestion
  useEffect(() => {
    if (hostHasStripe) {
      loadStripe();
    }
  }, [hostHasStripe]);
```

Reply:
I'm wondering if the step payment is the more suited to do that. Loading Stripe can take a few seconds and we don't want to slow people down. Maybe more suited to load it at the beginning of the flow. We'll see how that react live

Reply:
You mean the index? It's already in step payment.

Reply:
No, I mean the `loadStripe`